docs/adr/adr-009-telemetry.md

```diff
@@ -0,0 +1,226 @@
+
+Celestia Node needs deeper observability of each module and their components. The only integrated observability solution 
+we have is logging and there are two more options we need to explore from the observability triangle(tracing, metrics and logs).
+
+There are several priorities and "why"s we need deeper observability:
+* Analysis of the current p2p share exchange or "ShrEx" stack
+  * So we can evaluate real world Full Node reconstruction qualities, along with data availability sampling
+  * And adjust our roadmap accordingly.
+* Incentivized Testnet
+  * Tracking participants and validation that do task correctly
+  * So all participants provide to us valuable data/insight/traces that we can analyze and improve on
+* Monitoring dashboars
+  * For Celestia's own DA network infrastructure, e.g. DA Network Bootstrappers
+  * For the node operators
+* Extend debugging arsenal for the networking heavy DA layer
+  * Local development
+  * Issues found with Testground
+  * Production
+* Establishing metrics/data driven engineering culture for celestia-node devs
+  * Metrics and tracing allows extracting dry facts out of any software on its performance, liveness, bottlenecks,
+    regressions, etc., on whole system scale, so devs can reliably respond
+  * Basing on these, all the improvements can be proven with data _before_ and _after_ a change
+
+This ADR is intended to outline the decisions on how to proceed with:
+* Integration plan according to the priorities and the requirements
+* What observability tools/dependencies to integrate
+* Integration design into Celestia-Node for each observability option
+* A reference document explaining "whats" and "hows" during integration in some part of the codebase, e.g. new dev
+
+## Decisions
+
+### Plan
+
+The first "ShrEx" stack analysis priority is critical for Celestia project. The analysis results will tell us whether
+our current Full Node reconstruction qualities conforms to the main network requirements, subsequently affecting 
+the development roadmap of the celestia-node before the main network launch, therefore is a potential blocker to the 
+Basing on the former, the plan is focused on unblocking the reconstruction 
+story first and then proceed with steady covering of our codebase with traces for the complex codepaths as well as 
+metrics and dashboards for "measurables".
+
+Fortunately, the "ShrEx" analysis can be performed with _tracing_ only(more on that in [Tracing](./#Tracing)), so the
+decision for the celestia-node team is to cover with traces only the _necessary_ for the current "ShrEx" stack code as 
+the initial response to the ADR, leaving the rest to be integrated in the background for the devs in the team once they 
+are free as well as for the efficient bootstrapping into the code for the new devs.
+
+The next biggest priority - incentivized Testnet can be largely covered with traces as well. All participant will submit
```

Comment:
Replicating the style of Tooling/Dependencies here will make the reading much easier
Smt like:

### Plan

#### First Priority

#### Second Priority

Reply:
On it